ktor-io/posix/src/io/ktor/utils/io/ByteChannelSequentialBaseSharedStateNative.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.utils.io
+
+import io.ktor.utils.io.concurrent.*
+import io.ktor.utils.io.core.*
+import io.ktor.utils.io.core.internal.*
+
+internal actual class ByteChannelSequentialBaseSharedState actual constructor() {
+    actual var closed: Boolean by shared(false)
+
+    actual var readByteOrder: ByteOrder by shared(ByteOrder.BIG_ENDIAN)
+
+    actual var writeByteOrder: ByteOrder by shared(ByteOrder.BIG_ENDIAN)
+
+    actual var totalBytesRead: Long by shared(0L)
+
+    actual var totalBytesWritten: Long by shared(0L)
+
+    actual var closedCause: Throwable? by shared(null)
+
+    actual var lastReadAvailable: Int by shared(0)
+
+    actual var lastReadView: ChunkBuffer by shared(ChunkBuffer.Empty)
```

Comment:
Why not to use plain atomics on native, as those classes already splitted by js/jvm/native?
If it was a problem for JVM, may be it will also have perf problem for native, to create all those objects all time, and so on?

Reply:
It shouldn't be a big problem for native, but let's change them to atomics

Reply:
@e5l can you look at https://youtrack.jetbrains.com/issue/KTOR-1049#focus=Comments-27-4565293.0-0
Seems, that this change improves performance on native. I don't see any other code change, that can cause such perf improvement. Or do you know, what other can increase it?